base_geoengine/geo_db.py

```diff
@@ -18,22 +18,36 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
+"""Helper to setup Postgis"""
+from openerp.osv import osv
+from openerp.tools.translate import _
 
-from osv import fields, osv
+import logging
 
-from base_geoengine import geo_model
+logger = logging.getLogger('geoengine.sql')
 
 
-class SaleOrder(geo_model.GeoModel):
-    """Add geo_point to sale.order"""
-    _name = "sale.order"
-    _inherit = "sale.order"
-    _columns = {'geo_point' : fields.geo_related('partner_invoice_id',
-                                                 'geo_point',
-                                                 string='Sale order coordinate',
-                                                 type='geo_point',
-                                                 relation='res.partner.address',
-                                                 dim=2, # mandatory arg
-                                                 srid=900913)}# mandatory arg
-SaleOrder()
-
+def init_postgis(cursor):
+    """ Initialize postgis
+    """
+    cursor.execute("""
+        SELECT
+            tablename
+        FROM
+            pg_tables
+        WHERE
+            tablename='spatial_ref_sys';
+    """)
+    check = cursor.fetchone()
+    if check:
+        return {}
+    try:
+        cursor.execute("""
+        CREATE EXTENSION postgis;
+        CREATE EXTENSION postgis_topology;
+    """)
+    except Exception, exc:
+        raise osv.except_osv(
+            _('Error, Can not initialize spatial postgis function.'
+              ' Database user may have to be superuser and postgres/postgis '
+              'extentions and dev header have to be installed'), str(exc))
```

Comment:
Two things here, `_()` works by looking at the `context` variable in the local frame, if it's missing, there may not be a proper translation.

`str(exc)` might cause an `AsciiError` on non-english systems, use `ustr` from `openerp.tools` to avoid that.